c10/core/TensorOptions.h

```diff
@@ -507,6 +509,8 @@ inline TensorTypeId computeTensorTypeId(TensorOptions options) {
           return IDEEPTensorId();
         case DeviceType::HIP:
           return HIPTensorId();
+        case DeviceType::TPU:
+          return XLATensorId();
```

Comment:
is there a reason one of these is called TPU and one of them XLA Tensor?  Is it because in theory XLA tensors can run on other devices?  I guess the issue is that we have these one-to-one mappings so we wouldn't actually be able to differentiate in this way.  Do you mind calling this TPUTensorId or XLATPUTensorId or something?

Reply:
It's both theory and practice, we run our tests using the XLA CPU backend. `XLATensorId` is a better reflection of what this actually is, thus we like it better. Do you agree to keep it `XLATensorId`?

Reply:
Renamed the device type to `XLA`, let me know if it doesn't work.